modules/service/src/main/scala/lucuma/odb/graphql/mapping/MutationMapping.scala

```diff
@@ -750,24 +774,47 @@ trait MutationMapping[F[_]] extends Predicates[F] {
 
   private lazy val UpdateObservations: MutationField =
     MutationField("updateObservations", UpdateObservationsInput.binding(Path.from(ObservationType))) { (input, child) =>
-      services.useTransactionally {
 
-        val idSelect: Result[AppliedFragment] =
-          observationIdSelect(input.includeDeleted, input.WHERE, false)
+      // If any property other than grouping changes then disallow the edit for observations
+      // that are ongoing or completed.
+      def allowedStates(SET: ObservationPropertiesInput.Edit): Set[ObservationWorkflowState] =
+        if
+          SET.subtitle.isDefined            ||
+          SET.scienceBand.isDefined         ||
+          SET.posAngleConstraint.isDefined  ||
+          SET.targetEnvironment.isDefined   ||
+          SET.constraintSet.isDefined       ||
+          SET.timingWindows.isDefined       ||
+          SET.attachments.isDefined         ||
+          SET.scienceRequirements.isDefined ||
+          SET.observingMode.isDefined       ||
+          SET.existence.isDefined           ||
+          SET.observerNotes.isDefined
```

Comment:
This idea seems good to me.  I wonder if we should move this into the `ObservationPropertiesInput.Edit` itself to help keep up with it as stuff gets added.

Reply:
I had it there at first actually, but I wasn't sure it belonged there so I moved it. I may move it back.